CarpetX/src/schedule.cxx

```diff
@@ -1122,17 +1164,9 @@ int Initialise(tFleshConfig *config) {
     if (CCTK_EQUALS(timestep_choice, "timestep")) {
       cctkGH->cctk_delta_time = timestep;
     } else if (CCTK_EQUALS(timestep_choice, "dtfac")) {
-      CCTK_REAL mindx = 1.0 / 0.0;
-      for (const auto &patchdata : ghext->patchdata) {
-        const amrex::Geometry &geom = patchdata.amrcore->Geom(0);
-        const CCTK_REAL *restrict const dx = geom.CellSize();
-        CCTK_REAL mindx1 = 1.0 / 0.0;
-        for (int d = 0; d < dim; ++d)
-          mindx1 = fmin(mindx1, dx[d]);
-        mindx1 = ldexp(mindx1, -(int(patchdata.leveldata.size()) - 1));
-        mindx = fmin(mindx, mindx1);
-      }
-      cctkGH->cctk_delta_time = dtfac * mindx;
+      cctkGH->cctk_delta_time =
+          dtfac *
+          (ghext->use_subcycling() ? get_coarse_mindx() : get_finest_mindx());
     } else {
       abort();
```

Comment:
not your code but: better error reporting here would be nice. Or at least a comment that this is a bug in handling all possible cases of the keyword.